CMakeLists.txt

```diff
@@ -21,19 +30,19 @@ OPTION(EXPORT_BUILD
  "Add this build directory to CMake's user package registry.\
  Allows the package to be found without install." OFF)
 
-# Find common dependencies (used by multiple required modules)
+# Find all dependencies here, and ensure they have IMPORTED targets
 
 # Require Eigen 3.2.92, also called 3.3 beta-1, since it's in xenial
-FIND_PACKAGE(Eigen3 3.2.92 REQUIRED)
-FIND_PACKAGE(Boost 1.54.0 REQUIRED system filesystem)
-FIND_PACKAGE(PCL 1.8 REQUIRED
+FIND_PACKAGE(Eigen3 3.2.92)
```

Comment:
So now that it's not required...what will happen? I guess the module you're trying to include will require it, and it'll fail there?

Reply:
`WAVE_CHECK_MODULE` will cancel build of the dependent module.

Example output:
```
-- Building wave_utils
-- Building wave_geometry
-- Not building wave_containers: Requires MissingDep
-- Not building wave_benchmark: Requires wave::containers
-- Building wave_controls
-- Building wave_kinematics
-- Building wave_matching
-- Not building wave_vision: Requires wave::containers
-- Not building wave_optimization: Requires wave::vision
```

Reply:
Maybe I should make Eigen required anyway; it would be funny for the configure to succeed even though zero modules can be built.